Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -243,20 +244,35 @@ - (instancetype)initWithRequest:(NSURLRequest*)request
     };
 }
 
+// Helper function that executes a block on the client thread - used for client callbacks
+static void __dispatchClientCallback(NSURLProtocol_WinHTTP* protocol, void (^callbackBlock)()) {
+    if (protocol->_clientThread) {
+        [protocol performSelector:@selector(_invokeBlock:) onThread:protocol->_clientThread withObject:callbackBlock waitUntilDone:NO];
```

Comment:
1. They do inherit NSObject! It's awesome/terrifying all at the same time.
2. Meh, I left that comment a long time ago because I didn't think `-invoke` was valid.

I did just test it, though:

```
(duhowett-mac) ~/woc/block % cat main.m
#include <stdio.h>

int main() {
        void (^x)() = ^{ fprintf(stderr, "lol?\n"); };
        [x invoke];
        return 0;
}
(duhowett-mac) ~/woc/block % ./main
lol?
```